graveler/graveler.go

```diff
@@ -425,6 +428,9 @@ type StagingManager interface {
 	// List returns a ValueIterator for the given staging token
 	List(ctx context.Context, st StagingToken) (ValueIterator, error)
 
+	// IsDirty returns true iff staging area for the token isn't empty
+	IsDirty(ctx context.Context, st StagingToken) (bool, error)
```

Comment:
Why not use the List API?

Reply:
To avoid iterating and fetching unnecessary objects.

Reply:
if we will have an issue with the iterator we can optimize it